ra/SAPHana

```diff
@@ -1241,50 +1240,49 @@ function check_for_primary() {
                 # if 'actual_mode' is not available, fallback to 'mode'
-                if [ -z "$ini_mode" ]; then
-                    ini_mode=$(echo "$node_full_status" | awk -F= '$1=="mode" {print $2}')
+                if [ -z "$node_status" ]; then
+                    [[ "$node_full_status" =~ mode=([^$'\n']+) ]] && node_status=${BASH_REMATCH[1]}
                 fi
-                node_status="$ini_mode"
                 super_ocf_log info "ACT: Using getParameter.py as fallback - node_status=$node_status"
                 ;;
-           hU | * )
-                # DONE: PRIO1: Beginning from SAP HANA rev 112.03 -sr_state is not longer supported
-                node_full_status=$(HANA_CALL --timeout "$HANA_CALL_TIMEOUT" --cmd "$hdbState" 2>/dev/null )
-                node_status=$(echo "$node_full_status" | awk '$1=="mode:" {print $2}')
+            hU | * )
+                node_full_status=$(HANA_CALL --timeout "$HANA_CALL_TIMEOUT" --cmd "$hdbState --sapcontrol=1" 2>/dev/null )
+                [[ "$node_full_status" =~ mode=([^$'\n']+) ]] && node_status=${BASH_REMATCH[1]}
                 super_ocf_log debug "DBG: check_for_primary: node_status=$node_status"
                 ;;
-       esac
-       case "$node_status" in
-           primary )
-               rc=$HANA_STATE_PRIMARY
-               break;;
-           syncmem | sync | async )
-               rc=$HANA_STATE_SECONDARY
-               break;;
-           none ) # have seen that mode on second side BEFEORE we registered it as replica
-               rc=$HANA_STATE_STANDALONE
-               break;;
-           * )
-               super_ocf_log err "ACT: check_for_primary:  we didn't expect node_status to be: <$node_status>"
-               dump=$( echo $node_status | hexdump -C );
-               super_ocf_log err "ACT: check_for_primary:  we didn't expect node_status to be: DUMP <$dump>"
-               # TODO: Limit the runtime of hdbnsutil and use getParameter.py as fallback
-               # SAP_CALL
-               super_ocf_log debug "DEC: check_for_primary: loop=$i: node_status=$node_status"
-               # TODO: PRIO1: Maybe we need to keep the old value for P/S/N, if hdbnsutil just crashes
-       esac;
-       sleep 2
+        esac
+        case "$node_status" in
+            primary )
+                rc=$HANA_STATE_PRIMARY
+                break;;
+            syncmem | sync | async )
+                rc=$HANA_STATE_SECONDARY
+                break;;
+            none ) # have seen that mode on second side BEFEORE we registered it as replica
+                rc=$HANA_STATE_STANDALONE
+                break;;
+            * )
+                super_ocf_log err "ACT: check_for_primary:  we didn't expect node_status to be: <$node_status>"
+                dump=$( echo $node_status | hexdump -C );
+                super_ocf_log err "ACT: check_for_primary:  we didn't expect node_status to be: DUMP <$dump>"
+                ((i++))
```

Comment:
Where does this loop counter "i" come from? How is it initialized?
Should we define a local "i" with local i=0 ?

Reply:
Commit f44f729 from 20.09.2014 introduced the loop as `for in in 1 2 3 4 5 6 7 8 9` and added the debug message containing the loop counter `i` . With commit eb665c0 from 13.11.2015 the `for i` loop was exchanged with `for chkMethode in gP hU hU hU gP` , but the debug message was not changed even that there was no loop counter `i` anymore. And this missing `i` was not recognized all the last years till Peter's PR.
We can now change the debug message and remove the loop counter. Or we simply add the loop counter inside the loop as the PR proposes. As `i++` initializes the variable there is no need to explicitly initialize the variable. But if it is important to declare the loop count variable used in the debug message as local, we can declare and initialize this variable outside the loop